spec/hrr_rb_ssh/connection_spec.rb

```diff
@@ -586,6 +586,15 @@
         HrrRbSsh::Messages::SSH_MSG_CHANNEL_OPEN_FAILURE.new.encode channel_open_failure_message
       }
 
+      before(:each) do
+        # Disable the enhanced error messages introduced in newer Ruby versions which adds additional
+        # data to the expected 'undefined method `new' for nil:NilClass' exception
+        # https://github.com/ruby/error_highlight/tree/f3626b9032bd1024d058984329accb757687cee4#custom-formatter
+        if defined?(ErrorHighlight)
+          allow(ErrorHighlight::DefaultFormatter).to receive(:message_for).and_return('')
+        end
+      end
```

Comment:
In newer versions of ruby `e.message` contains extra metadata that we disable for this test:

Ruby 2.7
```
    176:   rescue => e
 => 177:     log_error { [e.backtrace[0], ": ", e.message, " (", e.class.to_s, ")\n\t", e.backtrace[1..-1].join("\n\t")].join }
    178:     recipient_channel = message[:'sender channel']
    179:     send_channel_open_failure recipient_channel, Messages::SSH_MSG_CHANNEL_OPEN_FAILURE::ReasonCode::SSH_OPEN_CONNECT_FAILED, e.message
    180:   end
    181: end

[1] pry(#<HrrRbSsh::Connection>)> e.message
NoMethodError: undefined method `message' for nil:NilClass
from (pry):1:in `rescue in channel_open'
Caused by NoMethodError: undefined method `new' for nil:NilClass
from /Users/user/Documents/code/hrr_rb_ssh/lib/hrr_rb_ssh/connection/channel.rb:39:in `initialize'
```

Ruby 3.1.1
```
    176:   rescue => e
 => 177:     log_error { [e.backtrace[0], ": ", e.message, " (", e.class.to_s, ")\n\t", e.backtrace[1..-1].join("\n\t")].join }
    178:     recipient_channel = message[:'sender channel']
    179:     send_channel_open_failure recipient_channel, Messages::SSH_MSG_CHANNEL_OPEN_FAILURE::ReasonCode::SSH_OPEN_CONNECT_FAILED, e.message
    180:   end
    181: end

[1] pry(#<HrrRbSsh::Connection>)> e.message
NoMethodError: undefined method `message' for nil:NilClass
from (pry):1:in `rescue in channel_open'
Caused by NoMethodError: undefined method `new' for nil:NilClass

        @channel_type_instance = ChannelType[@channel_type].new connection, self, message, socket, logger: logger
                                                           ^^^^
from /Users/user/Documents/code/hrr_rb_ssh/lib/hrr_rb_ssh/connection/channel.rb:39:in `initialize'
```